py2exe/hooks.py

```diff
@@ -805,3 +869,41 @@ def hook_wx(finder, module):
     when importing the full wx package
     """
     finder.excludes.append("wx.lib.pubsub")
+
+
+def hook_zmq(finder, module):
+    """pyzmq  >= 3.7.0 moved .dlls to a matplotlib.libs directory,
+    requiring that we override the '_delvewheel_init_patch_1_3_3' function
+    in mpl_tollkits so that those .dlls can be found in the same directory
+    as the executable.
+    """
+    import ast
+    from pkg_resources._vendor.packaging import version as pkgversion
+
+    import zmq
+    zmq_version = pkgversion.parse(zmq.__version__)
+
+    if zmq_version >= pkgversion.parse('23.0.0'):
```

Comment:
Also here I would encapsulate the whole hook in this `if` statement, so that the hook does nothing for older versions.